### DISCUSSION TITLE
Meson nasm compiler binary?

### DISCUSSION THREAD

**Discussion by ghost** [mesonbuild/meson#11854] [Q&A] [answered]
Hello,

I am here after facing some issues with the following document : https://mesonbuild.com/Release-notes-for-0-64-0.html#new-languages-nasm-and-masm;
I have declared that my project had nasm code, but for some reason, meson yells at me with : 
`ERROR: 'nasm' compiler binary not defined in cross or native file`,
I am not sure what this means, since nasm itself is a specific assembler.

**Comment by xclaesse** [comment, marked as answer]
When you're cross compiling you must define nasm in your cross file [binaries] section, like any other compiler.

But I think that's actually a Meson bug, nasm should not have to be defined in cross file because it's always the same compiler for all supported platforms, unlike gcc.

**Reply by xclaesse** [reply]
That PR should fix your problem, assuming you did install nasm on your build machine: https://github.com/mesonbuild/meson/pull/11855. Can you give it a try please?

**Reply by ghost** [reply]
> That PR should fix your problem, assuming you did install nasm on your build machine: #11855. Can you give it a try please?

your branch seems to have fixed the problem
